eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/annotation/load/VariantAnnotationMongoItemWriter.java

```diff
@@ -0,0 +1,168 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Write a list of {@link VariantAnnotation} into MongoDB
+ *
+ * A new annotation is added in the existing document.
+ * In case of two annotations (or more) in the same variant the other annotations are appended:
+ *
+ * 20_63963_G/A	20:63963	A	ENSG00000178591	ENST00000382410	Transcript	upstream_gene_variant	-	-	-	-	-	-	DISTANCE=4388;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=protein_coding;CANONICAL=YES;CCDS=CCDS12989.2;ENSP=ENSP00000371847;SWISSPROT=DB125_HUMAN;TREMBL=B2R4E8_HUMAN;UNIPARC=UPI00001A36DE
+ * 20_63963_G/A	20:63963	A	ENSG00000178591	ENST00000608838	Transcript	upstream_gene_variant	-	-	-	-	-	-	DISTANCE=3928;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=processed_transcript
+ *
+ * Will be:
+ *
+ * "annot" : {
+ *  "ct" : [
+ *      { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000382410", "codon" : "-", "strand" : "+", "bt" : "protein_coding", "aaChange" : "-", "so" : [ 1631 ] },
+ *      { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000608838", "codon" : "-", "strand" : "+", "bt" : "processed_transcript", "aaChange" : "-",
+ *  "so" : [ 1631 ] } ],
+ *  "xrefs" : [
+ *      { "id" : "DEFB125", "src" : "HGNC" },
+ *      { "id" : "ENST00000382410", "src" : "ensemblTranscript" },
+ *      { "id" : "ENST00000608838", "src" : "ensemblTranscript" },
+ *      { "id" : "ENSG00000178591", "src" : "ensemblGene"
+ */
+public class VariantAnnotationMongoItemWriter extends MongoItemWriter<VariantAnnotation> {
+    private static final Logger logger = LoggerFactory.getLogger(VariantAnnotationMongoItemWriter.class);
+
+    private MongoOperations mongoOperations;
+    private String collection;
+
+    public VariantAnnotationMongoItemWriter(MongoOperations mongoOperations) {
+        this.mongoOperations = mongoOperations;
+    }
+
+    @Override
+    public void setCollection(String collection) {
+        super.setCollection(collection);
+        this.collection = collection;
+    }
+/*    public VariantAnnotationMongoItemWriter() {
+        this.template = mongoTemplate();
+    }
+
+    public MongoTemplate mongoTemplate() {
+        MongoTemplate mongoTemplate;
+        try {
+            //mongoTemplate = new MongoTemplate(new MongoClient(), pipelineOptions.getString(VariantStorageManager.DB_NAME));
+            mongoTemplate = new MongoTemplate(new MongoClient(), "variants");
+        } catch (UnknownHostException e) {
+            throw new RuntimeException("Unable to initialize MongoDB", e);
+        }
+        return mongoTemplate;
+    }*/
+
+    @Override
+    protected void doWrite(List<? extends VariantAnnotation> variantAnnotations) {
+        //// TODO: 30/06/2016 move in constructor?
+        DBObjectToVariantAnnotationConverter converter = new DBObjectToVariantAnnotationConverter();
+
+        for (VariantAnnotation variantAnnotation : variantAnnotations) {
```

Comment:
And PRE-merging the VariantAnnotation before the query... Yes! I will look into this